packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -257,7 +265,7 @@ def _is_langchain_chain(self, name):
         if not acceptable_name.match(name):
             return False
 
-        ipython = get_ipython()
+        ipython = self.shell
```

Comment:
@krassowski 
I am not deeply familiar with magics, curious if this change has any implications or is a drop-in replacement for the existing statement?

Reply:
It is drop-in replacement, except that it now can be properly tested and used outside of IPython interpreter. There is a reason why `Magics` constructor receives the `shell` argument. `get_ipython()` should never be used in magics.